### ISSUE TITLE
Type parameter of GetLogger/GetCurrentClassLogger isn't clear

### ISSUE THREAD

**Issue by 304NotModified** [NLog/NLog#961]
We can create a Logger of a specified type in `LogManager`, `LogFactory` and `LogFactory<TLogger>`

With these methods:

``` c#
        /// <summary>
        /// Gets the specified named logger.
        /// </summary>
        /// <param name="name">Name of the logger.</param>
        /// <param name="loggerType">The logger class. The class must inherit from <see cref="Logger" />.</param>
        /// <returns>The logger reference. Multiple calls to <c>GetLogger</c> with the same argument aren't guaranteed to return the same logger reference.</returns>
        /// <remarks>The generic way for this method is <see cref="LogFactory{loggerType}.GetLogger(string)"/></remarks>
        [CLSCompliant(false)]
        public static Logger GetLogger(string name, Type loggerType)

        /// <summary>
        /// Gets the logger named after the currently-being-initialized class.
        /// </summary>
        /// <param name="loggerType">The logger class. The class must inherit from <see cref="Logger" />.</param>
        /// <returns>The logger.</returns>
        /// <remarks>This is a slow-running method. 
        /// Make sure you're not doing this in a loop.</remarks>
        [CLSCompliant(false)]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public static Logger GetCurrentClassLogger(Type loggerType)
        {
            return factory.GetLogger(GetClassFullName(), loggerType);            
        }

```

It's unclear to people what `loggerType` does, as seen in #960 and https://github.com/NLog/NLog/issues/263. 

What it does:

```
1. It expects it inherits from `Logger`
2. Instead of creating an instance of `Logger`, it will create an instance of `loggerType`
```

It can be useful when custom Loggers are created.

What we need to improve:
- [x] The error message
- [x] the xml doc 
- [x] The documentation in code
- [ ] The documentation in the wiki, e.g https://github.com/NLog/NLog/wiki/Tutorial#expose-logger-to-sub-classes

**Comment by 304NotModified** [comment]
working at this in code.

**Comment by snakefoot** [comment]
Resolved by #3585 with NLog v4.6.8.

Notice NLog v5.2 marks `GetLogger(string name, Type loggerType)` as obsolete and redirects to `GetLogger<T>(string name) where T : Logger`
